source/OptimaJet.DWKit.StarterApplication/Forms/Projects/UpdateForm.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Linq;
+using JsonApiDotNetCore.Data;
+using JsonApiDotNetCore.Internal;
+using Microsoft.EntityFrameworkCore;
+using Optimajet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+using OptimaJet.DWKit.StarterApplication.Services;
+
+namespace OptimaJet.DWKit.StarterApplication.Forms.Projects
+{
+    public class UpdateForm : BaseProjectForm
+    {
+        public UserRepository UserRepository { get; set; }
+        public GroupRepository GroupRepository { get; set; }
+        public IEntityRepository<Organization> OrganizationRepository { get; set; }
+        public ICurrentUserContext CurrentUserContext { get; }
+        public ProjectRepository ProjectRepository { get; set; }
+        public UpdateForm(
+            UserRepository userRepository,
+            GroupRepository groupRepository,
+            ICurrentUserContext currentUserContext,
+            IEntityRepository<Organization> organizationRepository,
+            ProjectRepository projectRepository) : base()
+        {
+            UserRepository = userRepository;
+            GroupRepository = groupRepository;
+            CurrentUserContext = currentUserContext;
+            OrganizationRepository = organizationRepository;
+            ProjectRepository = projectRepository;
+        }
+
+        public bool IsValid(int id, Project project)
+        {
+            var result = true;
+            if ((project.OrganizationId != VALUE_NOT_SET)
+                || (project.OwnerId != VALUE_NOT_SET)
+                || (project.GroupId != VALUE_NOT_SET))
+            {
+                var original = ProjectRepository.Get()
+                                  .Where(p => p.Id == id)
+                                  .Include(p => p.Organization)
+                                  .Include(p => p.Owner)
+                                          .ThenInclude(u => u.GroupMemberships)
+                                              .ThenInclude(gm => gm.Group)
+                                  .Include(p => p.Owner)
+                                          .ThenInclude(u => u.OrganizationMemberships)
+                                               .ThenInclude(om => om.Organization)
+                                  .Include(p => p.Group)
+                                     .ThenInclude(g => g.Owner)
+                                  .FirstOrDefaultAsync().Result;
+                Owner = original.Owner;
+                Organization = original.Organization;
+                Group = original.Group;
+                // Set the fields to what would be the updated versions
+                // Fields that don't change, get their value from the original
+                if (project.OrganizationId != VALUE_NOT_SET)
+                {
+                    Organization = OrganizationRepository.Get()
```

Comment:
The Organization is being updated (!= VALUE_NOT_SET) therefore the organization to use to validate in the base project must come from the one specified by the update organization id rather than the one from the orginal record.